python/ray/tests/kuberay/test_autoscaling_config.py

```diff
@@ -336,6 +389,50 @@ def _fetch_ray_cr_from_k8s(self) -> Dict[str, Any]:
             assert out == {"ok-key": "ok-value"}
 
 
+TPU_PARAM_ARGS = ",".join(
+    [
+        "ray_cr_in",
+        "expected_num_tpus",
+    ]
+)
+TPU_TEST_DATA = (
+    pytest.param(
+        get_basic_ray_cr(),
+        4,
+        id="tpu-k8s-resource-limits",
+    ),
+    pytest.param(
+        _get_ray_cr_with_tpu_custom_resource(),
+        4,
+        id="tpu-custom-resource-ray-start-params",
+    ),
+    pytest.param(
+        _get_ray_cr_with_no_tpus(),
+        0,
+        id="no-tpus-requested",
+    ),
```

Comment:
Can we also add a test that doesn't specify the container's resource limit but sets the `rayStartParams`?

Reply:
Done.